reader.go

```diff
@@ -0,0 +1,114 @@
+package cbzip2
+
+/*
+#cgo LDFLAGS: -lbz2
+#cgo CFLAGS: -Werror=implicit
+
+#include <bzlib.h>
+*/
+import "C"
+import (
+	"io"
+	"unsafe"
+)
+
+type Reader struct {
+	r      io.Reader
+	bz     *C.bz_stream
+	in     []byte
+	skipIn bool
+	err    error
+}
+
+// NewReader returns an io.Reader. Reads from this are read from the
```

Comment:
It's _technically_ an `io.ReadCloser`.  I see you changed the return type to `*Reader` too.  Shouldn't make a big difference either way on this one though.